crates/cxx-qt-build/src/dependencies.rs

```diff
@@ -0,0 +1,301 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! This modules contains utilities for specifying dependencies with cxx-qt-build.
+
+use serde::{Deserialize, Serialize};
+
+use std::collections::{HashMap, HashSet};
+use std::path::{Path, PathBuf};
+
+/// When generating a library with cxx-qt-build, the library may need to export certain flags or headers.
+/// These are all specified by this Interface struct, which should be passed to the [crate::CxxQtBuilder::library] function.
+pub struct Interface {
```

Comment:
does this need to be non exhaustive to avoid ABI breaks ? (eg if we are adding more fields) or does it work as the have to go through methods like default and cannot access the internals?

Reply:
The internals are `pub(crate)`, so you can't access them from outside :)